index.js

```diff
@@ -319,36 +317,46 @@ const captureWebsite = async (input, options) => {
 	}
 
 	if (screenshotOptions.fullPage) {
-		// Get the height of the rendered page
-		const bodyHandle = await page.$('body');
-		const bodyBoundingHeight = await bodyHandle.boundingBox();
-		await bodyHandle.dispose();
-
-		// Scroll one viewport at a time, pausing to let content load
-		const viewportHeight = viewportOptions.height;
-		let viewportIncrement = 0;
-		while (viewportIncrement + viewportHeight < bodyBoundingHeight) {
-			const navigationPromise = page.waitForNavigation({waitUntil: 'networkidle0'});
-			/* eslint-disable no-await-in-loop */
-			await page.evaluate(_viewportHeight => {
-				/* eslint-disable no-undef */
-				window.scrollBy(0, _viewportHeight);
-				/* eslint-enable no-undef */
-			}, viewportHeight);
-			await navigationPromise;
-			/* eslint-enable no-await-in-loop */
-			viewportIncrement += viewportHeight;
+		const autoScroll = async () => {
+			// Scroll and check if page is at bottom
+			const isBottom = await page.evaluate(() => {
+				window.scrollBy(0, window.innerHeight);
+				return window.scrollY >= document.body.clientHeight - window.innerHeight;
+			});
+
+			// Wait for idle event loop to ensure client-side rendering is complete
+			await page.evaluate(async () => {
+				return new Promise(resolve => {
+					window.requestIdleCallback(resolve, {timeout: 100});
+				});
+			});
+
+			// Ensure images are done loading
+			await page.waitForFunction(() => [...document.images].every(element => element.complete));
+
+			return !isBottom;
+		};
+
+		while (await autoScroll()) { /* eslint-disable-line no-await-in-loop */
```

Comment:
Should this have a timeout? Otherwise, it would just go on forever on sites with infinite auto-loading.